runtests.py

```diff
@@ -805,6 +813,8 @@ def build_tests(self, test_class, path, workdir, module, module_path, expect_err
         warning_errors = 'werror' in tags['tag']
         expect_warnings = 'warnings' in tags['tag']
 
+        extra_directives_list = [None]
```

Comment:
I think this would simplify a couple of places where it's used. You can still do a truth test on it, but you don't have to any more.
```suggestion
        extra_directives_list = [{}]
```